packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -299,18 +300,22 @@ class AnalyzeCommand extends FlutterCommand {
   Map<String, List<AnalysisError>> analysisErrors = <String, List<AnalysisError>>{};
   Stopwatch analysisTimer;
   int lastErrorCount = 0;
+  Status analysisStatus;
 
   void _handleAnalysisStatus(AnalysisServer server, bool isAnalyzing) {
     if (isAnalyzing) {
+      analysisStatus?.cancel();
+
       if (firstAnalysis) {
-        printStatus('Analyzing ${path.basename(Directory.current.path)}...');
+        analysisStatus = logger.startProgress('Analyzing ${path.basename(Directory.current.path)}...');
```

Comment:
ah, the name it prints is wrong. It's assuming that you're in the root of the flutter repo